bayesian/utility/v2/component_analyses.py

```diff
@@ -241,26 +290,29 @@ def get_known_unknown_pkgs(
     return stack_recommendation, unknown_pkgs
 
 
-def get_known_clair_pkgs(
-        ecosystem: str, graph_response: Dict) -> Tuple[List[Dict], Set[Package]]:
+def check_vulnerable_package(version, version_string):
+    """Check if the request version available in vulnerable_versions."""
+    if(version in version_string.split(",")):
+        return True
+    return False
+
+
+def get_known_pkgs(graph_response: Dict, packages_list: Dict) -> List[Dict]:
     """Analyse Known and Unknown Packages."""
-    stack_recommendation = []
+    for temp in packages_list:
+        temp["vulnerabilities"] = []
     for package in graph_response.get('result', {}).get('data'):
-        pkg_name = package.get('package').get('name', [''])[0]
-        pkg_version = package.get('version').get('version', [''])[0]
-        pkg_recomendation = CABatchResponseBuilder(ecosystem).generate_recommendation(
-            package, pkg_name, pkg_version)
-        vulnerabilities_array = "No vulnerabilities found for package"
-        if "vulnerability" in pkg_recomendation:
-            vulnerabilities_array = pkg_recomendation["vulnerability"]
-        custom_pkg_recomendation = {
-            "package": pkg_recomendation["package"],
-            "version": pkg_recomendation["version"],
-            "vulnerabilities": vulnerabilities_array
-        }
-        stack_recommendation.append(custom_pkg_recomendation)
-
-    return stack_recommendation
+        for response_package in packages_list:
```

Comment:
if the number of packages in the input are high, this n*n processing will get really delayed response

Reply:
i still feel a use of map will help run things extremely fast. dont worry about space. python will trigger dc as soon as the object ref is removed

Reply:
Also, in case you get multiple vuln for the same package, you will be wasting so much time traversing through the list again and again to get to the same version and package name. This will be a very common usecase of having multiple vuln for the same package/version

Reply:
yup, added new logic